cdap-ui/app/cdap/components/CaskWizards/MicroserviceUpload/ConfigureStep/index.js

```diff
@@ -0,0 +1,201 @@
+const mapStateToMicroserviceMemoryProps = (state) => {
+  return {
+    value: state.configure.memory,
+    type: 'number',
+    min: '1',
+    placeholder: T.translate('features.Wizard.MicroserviceUpload.Step4.memoryPlaceholder')
+  };
+};
+const mapStateToMicroserviceThresholdProps = (state) => {
+  return {
+    value: state.configure.ethreshold,
+    type: 'number',
+    min: '1',
+    placeholder: T.translate('features.Wizard.MicroserviceUpload.Step4.thresholdPlaceholder')
+  };
+};
+const mapStateToConfigureStepSummaryProps = (state) => {
+  return {
+    instanceName: state.general.instanceName,
+    instances: state.configure.instances,
+    vcores: state.configure.vcores,
+    memory: state.configure.memory,
+    ethreshold: state.configure.ethreshold,
+  };
+};
+
+
+const mapDispatchToMicroserviceInstancesProps = (dispatch) => {
+  return {
+    onChange: (e) => {
+      dispatch({
+        type: MicroserviceUploadActions.setInstances,
+        payload: {instances: e.target.value}
+      });
+    }
+  };
+};
+const mapDispatchToMicroserviceVCoresProps = (dispatch) => {
+  return {
+    onChange: (e) => (dispatch({
+      type: MicroserviceUploadActions.setVCores,
+      payload: {vcores: e.target.value}
+    }))
+  };
+};
+const mapDispatchToMicroserviceMemoryProps = (dispatch) => {
+  return {
+    onChange: (e) => (dispatch({
+      type: MicroserviceUploadActions.setMemory,
+      payload: {memory: e.target.value}
+    }))
+  };
+};
+const mapDispatchToMicroserviceThresholdProps = (dispatch) => {
+  return {
+    onChange: (e) => (dispatch({
+      type: MicroserviceUploadActions.setThreshold,
+      payload: {ethreshold: e.target.value}
+    }))
+  };
+};
+
+let Summary = ({instanceName, instances, vcores, memory, ethreshold}) => {
+  let instancesWithCount = T.translate('features.Wizard.MicroserviceUpload.Step4.summary.count.instances', { context: instances });
+  let vcoresWithCount = T.translate('features.Wizard.MicroserviceUpload.Step4.summary.count.vcores', { context: vcores });
+  let memoryWithCount = T.translate('features.Wizard.MicroserviceUpload.Step4.summary.count.memory', { context: memory });
+  let ethresholdWithCount = T.translate('features.Wizard.MicroserviceUpload.Step4.summary.count.ethreshold', { context: ethreshold });
+  return (
+    <span>
+      {T.translate('features.Wizard.MicroserviceUpload.Step4.summary.text', { instanceName, instancesWithCount, vcoresWithCount, memoryWithCount, ethresholdWithCount })}
+    </span>
+  );
+};
+
+Summary.propTypes = {
+  instanceName: PropTypes.string,
+  instances: PropTypes.number,
+  vcores: PropTypes.number,
+  memory: PropTypes.number,
+  ethreshold: PropTypes.number
+};
+
+const InputMicroserviceInstances = connect(
+  mapStateToMicroserviceInstancesProps,
+  mapDispatchToMicroserviceInstancesProps
+)(InputWithValidations);
+const InputMicroserviceVCores = connect(
+  mapStateToMicroserviceVCoresProps,
+  mapDispatchToMicroserviceVCoresProps
+)(InputWithValidations);
```

Comment:
I am not sure if `Input` vs `InputWithValidations` makes any difference here. I am not even sure why we are using it for text fields. We are not passing in `validationError` as prop and no error is ever displayed. The original intention in wizard is in future if we get an error per field we display them but I don't think that is happening.